straxen/corrections_services.py

```diff
@@ -0,0 +1,222 @@
+            self.password = straxen.get_secret('rundb_password')
+
+        if host is None:
+            mongo_connections = [default_mongo_url, *backup_mongo_urls]
+            for url in mongo_connections:
+                try:
+                    mongo_url = f'mongodb://{url}'
+
+                    self.interface = strax.CorrectionsInterface(
+                        host=mongo_url,
+                        username=self.username,
+                        password=self.password,
+                        database_name='corrections')
+                    break
+                except pymongo.errors.ServerSelectionTimeoutError:
+                    warn(f'Cannot connect to to Mongo url: {url}')
+                    if url == mongo_connections[-1]:
+                        raise pymongo.errors.ServerSelectionTimeoutError(
+                            'Cannot connect to any Mongo url')
+        else:
+            self.interface = strax.CorrectionsInterface(
+                    host=self.host,
+                    username=self.username,
+                    password=self.password,
+                    database_name='corrections')
+
+    def get_corrections_config(self, run_id, correction=None, config_model=None):
+        """
+        Get context configuration for a given correction
+        :param run_id: run id from runDB
+        :param correction: correction's name (str type)
+        :param config_model: configuration model (dict type)
+        :return: correction value(s)
+        """
+
+        if not isinstance(config_model, tuple):
+            raise ValueError(f'config_model {config_model} must be a tuple')
+        if len(config_model) == 3:
+            model_type, global_version, xenon1t = config_model
+        elif len(config_model) == 2:
+            model_type, global_version = config_model
+            xenon1t = False
+
+        if correction == 'pmt_gains':
+            if model_type == 'to_pe_model':
+                to_pe = self.get_pmt_gains(run_id, global_version, xenon1t)
+                return to_pe
+
+            elif model_type == 'to_pe_constant':
+                n_tpc_pmts = straxen.n_tpc_pmts
+                if xenon1t:
+                    n_tpc_pmts = 248
+                cte_value = global_version
+                to_pe = np.ones(n_tpc_pmts, dtype=np.float32) * cte_value
+                return to_pe
+
+            else:
+                raise ValueError(f'{model_type} not implemented')
+
+        elif correction == 'elife':
+            if model_type == 'elife_model':
+                elife = self.get_elife(run_id, global_version, xenon1t)
+                return elife
+            elif model_type == 'elife_constant':
+                cte_value = global_version
+                elife = float(cte_value)
+                return elife
+        else:
+            raise ValueError(f'{corection} not found')
+
+    def get_elife(self, run_id, global_version='v1', xenon1t=False):
```

Comment:
Hej Aaron, I looked through `get_elife` and `get_pmt_gains` and they both look very very similar. I think this will also be the case for the not yet implemented functions like `get_lce` and `get_fdc` so I was wondering if we could not be a bit smarter and just have a single function for all of those to avoid this copy and paste code. 
At the moment I do not see why this should not be possible. In case we have to change something on the straxen side (e.g. the format strax expects the PMT gains to be) no problem we can do that.

Reply:
I have done that, thanks for the suggestion